packages/server/events-processing-platform/notifications/postmark.go

```diff
@@ -43,18 +46,44 @@ type PostmarkEmailAttachment struct {
 }
 
 type PostmarkProvider struct {
-	Client *postmark.Client
-	log    logger.Logger
+	Client     *postmark.Client
+	log        logger.Logger
+	repository *repository.Repositories
 }
 
-func NewPostmarkProvider(log logger.Logger, serverToken string) *PostmarkProvider {
+func NewPostmarkProvider(log logger.Logger, repo *repository.Repositories) *PostmarkProvider {
 	return &PostmarkProvider{
-		Client: postmark.NewClient(serverToken, ""),
-		log:    log,
+		Client:     postmark.NewClient("", ""),
+		log:        log,
+		repository: repo,
 	}
 }
 
-func (np *PostmarkProvider) SendNotification(ctx context.Context, postmarkEmail PostmarkEmail, span opentracing.Span) error {
+func (np *PostmarkProvider) SetClientToken(tenant string) error {
```

Comment:
please return the token as a string and pass it below. please don't set np.Client.ServerToken